Floofbot/Modules/UserAssignableRoleCommands.cs

```diff
@@ -0,0 +1,257 @@
+﻿using System;
+using Discord;
+using System.Threading.Tasks;
+using Discord.Commands;
+using System.Text.RegularExpressions;
+using Floofbot.Services.Repository;
+using Floofbot.Services.Repository.Models;
+using System.Linq;
+using Discord.Addons.Interactive;
+using Discord.WebSocket;
+using Serilog;
+
+namespace Floofbot.Modules
+{
+    [Summary("Join roles that the administrator have allowed you to")]
+    public class UserAssignableRoleCommands : InteractiveBase
+    {
+        private FloofDataContext _floofDb;
+        private readonly Discord.Color EMBED_COLOUR = new Discord.Color((uint)new Random().Next(0x1000000));
+        public UserAssignableRoleCommands(FloofDataContext floofDb)
+        {
+            _floofDb = floofDb;
+        }
+
+        private SocketRole GetRole(string roleName, SocketGuild guild)
+        {
+            foreach (SocketRole r in guild.Roles)
+            {
+                if (r.Name.ToLower() == roleName)
+                    return r;
+            }
+            return null;
+        }
+        private bool CheckRoleEntryExists(SocketRole role, SocketGuild guild)
+        {
+            // checks if a word exists in the filter db
+            bool isRoleInDb = _floofDb.UserAssignableRoles.AsQueryable().Where(r => r.ServerId == guild.Id).Where(r => r.RoleId == role.Id).Any();
+            return isRoleInDb;
+        }
+
+        [Summary("Join a role that the server administrators allow you to")]
+        [Command("iam")]
+        public async Task IAm([Summary("role name")][Remainder]string roleName)
+        {
+            if (roleName == null)
```

Comment:
I think you might need to change the line above to `[Remainder]string roleName=null` for this to work, or maybe it has to use the empty string `""` as the default parameter instead.

Reply:
Set default value to null